config/software/ruby.rb

```diff
@@ -93,6 +93,8 @@
     if ohai['platform_version'].to_f >= 5.11
       patch source: "ruby-solaris-linux-socket-compat.patch", plevel: 1
     end
+  elsif solaris2? && version.to_f >= 1.9
```

Comment:
What happens if this patch is applied on Ruby 2.x? Perhaps we should guard like this:

```ruby
solaris2? && version =~ /^1.9/
```